### PR TITLE
Add support tools

### PR DESCRIPTION
**Pull request by naoa**
Add tools param

https://docs.cohere.com/reference/chat
https://docs.cohere.com/docs/tool-use

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,6 +64,34 @@ end
...
 `force_encoding` is preferred to avoid JSON parsing issue when Cohere returns emoticon.
 
+`chat` supports Tool use (function calling).
+
+```ruby
+tools = [
+   {
+     name: "query_daily_sales_report",
+     description: "Connects to a database to retrieve overall sales volumes and sales information for a given day.",
+     parameter_definitions: {
```
Comment (andreibondarev) [inline README.md:74]: Why did they have to call this key differently from OpenAI and Google Gemini? 😭😭😭

**Comment by andreibondarev** [comment]
@naoa Are you using Langchain.rb by any chance?

**Comment by naoa** [comment]
Thanks for merging!
I haven't used it yet, but it looks like a cool tool so I might try it next time!

**Comment by andreibondarev** [comment]
@naoa I'd take a look at the [Assistants](https://github.com/andreibondarev/langchainrb?tab=readme-ov-file#assistants) if you have a need for them. I've been trying [to figure out](https://github.com/andreibondarev/langchainrb/issues/543) how to best implement it to use different LLMs (any LLM that support functions calling).

**Pull request merged**